bin/install

```diff
@@ -1,115 +1,137 @@
-
-test -z ${OMF_REPO_URI+_}     && OMF_REPO_URI="https://github.com/oh-my-fish/oh-my-fish"
-test -z ${OMF_REPO_BRANCH+_}  && OMF_REPO_BRANCH="master"
-
-die() {
-  echo "$1" && exit 1
-}
-
-is_installed() {
-  type "$1" >/dev/null 2>&1
-}
-
-omf_create_fish_config() {
-  local fish_config_file=$1
-  mkdir -p $(dirname "${fish_config_file}")
-  touch "${fish_config_file}"
-}
-
-omf_install() {
-  echo "Resolving Oh My Fish path → ${OMF_PATH}"
-  test -d "${OMF_PATH}" && die "Existing installation detected, aborting"
-  local git_uri="$(echo ${OMF_REPO_URI} | sed 's/\.git//').git"
-
-  echo "Cloning Oh My Fish → ${git_uri}"
-  if ! git clone -q --depth 1 -b "${OMF_REPO_BRANCH}" "${git_uri}" "${OMF_PATH}"; then
-    echo "Is 'git' installed?"
-    die "Could not clone the repository → ${OMF_PATH}:${OMF_REPO_BRANCH}"
-  fi
-
-  local git_rev=$(git --git-dir ${OMF_PATH}/.git --work-tree ${OMF_PATH} rev-parse HEAD) >/dev/null 2>&1
-  local git_upstream=$(git --git-dir ${OMF_PATH}/.git --work-tree ${OMF_PATH} config remote.upstream.url)
-
-  if [ -z "${git_upstream}" ]; then
-    git --git-dir ${OMF_PATH}/.git --work-tree ${OMF_PATH} remote add upstream ${git_uri}
-  else
-    git --git-dir ${OMF_PATH}/.git --work-tree ${OMF_PATH} remote set-url upstream ${git_uri}
-  fi
+#   OMF_REPO_URI      Oh My Fish source git repository
+#   OMF_REPO_BRANCH   Oh My Fish source repository branch
+
+
+set -q XDG_DATA_HOME;   or set XDG_DATA_HOME "$HOME/.local/share"
+set -q XDG_CONFIG_HOME; or set XDG_CONFIG_HOME "$HOME/.config"
+
+set -q FISH_CONFIG;     or set FISH_CONFIG "$XDG_CONFIG_HOME/fish"
+
+set -q OMF_PATH;        or set OMF_PATH "$XDG_DATA_HOME/omf"
+set -q OMF_CONFIG;      or set OMF_CONFIG "$XDG_CONFIG_HOME/omf"
+
+set -q OMF_REPO_URI;    or set OMF_REPO_URI "https://github.com/oh-my-fish/oh-my-fish"
+set -q OMF_REPO_BRANCH; or set OMF_REPO_BRANCH "master"
 
-  echo "Oh My Fish revision id → ${git_rev}"
-  test -z ${FISH_CONFIG+_} && FISH_CONFIG="${XDG_CONFIG_HOME}/fish"
+set -l OMF_FISH_MIN_VER 2 1 0
 
-  local fish_config_file="${FISH_CONFIG}/config.fish"
+function available -a name
+  type "$name" >/dev/null 2>&1
+end
 
-  if [ -e "${FISH_CONFIG}/config.fish" ]; then
-    local timestamp=$(date +%s)
-    local fish_config_bk="${FISH_CONFIG}/config.${timestamp}.copy"
+function report -a what message
+  switch $what
+  case 'progress'
+    set_color yellow
+  case 'success'
+    set_color green
+  case 'error'
+    set_color red
+    printf "$message\n"
+    exit 1
+  end
 
-    echo "Found existing 'fish' configuration → ${fish_config_file}"
-    echo "Writing back-up copy → ${fish_config_bk}"
+  printf "$message\n"
+end
 
-    cp "${fish_config_file}" "${fish_config_bk}" >/dev/null 2>&1
-    test $? -ne 0 && die "Writing back-up copy failed, error code → ${?}"
+function fish_version_compatible
+  set -l MAJOR (echo $FISH_VERSION | cut -d. -f1)
+  set -l MINOR (echo $FISH_VERSION | cut -d. -f2)
```

Comment:
see [fish 2.0.0](https://github.com/fish-shell/fish-shell/commit/a10fa3a20b634c033476be15769bff1a37632f3f)
The env `$FISH_VERSION` is only available from 2.0.0+.
So you have to use `fish -v`.
Or maybe you can use `$version` present since [1.21.12](https://github.com/fish-shell/fish-shell/commit/f771fce892394ba877141416d07d930bc7456b2d)

`fish -v` is the only thing which has been immutable since the beginning.